pkg/controller/controller.go

```diff
@@ -20,25 +24,33 @@ var (
 )
 
 func NewManager(cfg *rest.Config, options manager.Options) (manager.Manager, error) {
-	mgr, err := manager.New(cfg, options)
-	if err != nil {
-		return nil, err
-	}
-
-	controllerLog.Info("Registering Components.")
+	options.Scheme = runtime.NewScheme()
 
 	// pretty sure this is there by default but we will be explicit like build-service
-	if err := k8sscheme.AddToScheme(mgr.GetScheme()); err != nil {
+	if err := k8sscheme.AddToScheme(options.Scheme); err != nil {
 		return nil, err
 	}
 
-	if err := v1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+	if err := v1alpha1.AddToScheme(options.Scheme); err != nil {
 		return nil, err
 	}
 
-	if err := pipelinev1beta1.AddToScheme(mgr.GetScheme()); err != nil {
+	if err := pipelinev1beta1.AddToScheme(options.Scheme); err != nil {
 		return nil, err
 	}
+	options.NewCache = cache.BuilderWithOptions(cache.Options{
```

Comment:
ok I realized something later tonight about my comment and I took another look, where my prior comments missed a couple of things ... I believe I now noticed that
- there is a 1 to 1 correspondence between artifactbuildrequests and taskruns, the 1 to 1 between dependencybuilds and pipelineruns based on the use of create in the artifactbuidlrequests and dependencybuild reconcilers
- but I see you doing lists in your pipelinerun and taskrun reconcilers to find their taskruns and pipelineruns

If I have that 1 to 1 correspondence right, I think we need to adjust this a little, though not in the way I mentioned in https://github.com/redhat-appstudio/jvm-build-service/pull/33#discussion_r873068039

Essentially, we have two labels:
1) a label with a constant k/v pair (where maybe the empty string for the value is OK, but if not just pick a non empty constant value) to help filter the pipelinerun and taskrun reconcilers to those associated with artifactbuildrequests and dependencybuilds
2) a second label where the value if the namespace/name needed so you use a Get instead of a List in the pipelinerun and taskrun reconcilers

Reply:
Are you saying that I should put the name of the ABR in a Label on the TaskRun and use that to lookup the ABR directly? 

The issue with this is that label values are much more restrictive than the naming rules, so the generated ABR names currently don't fit into a value. I could shrink the names somehow and enforce that they meet the label rules, but some human readability may be lost for long GAV's.

My solution here was to generate an artificial ID and put it in a label on both resources, and then use List to update the resources, but I can shrink the names if you want. 

Note that I already have two labels, one for correlation and empty one for the filtering like you suggest.